src/vfs2.c

```diff
@@ -826,9 +904,15 @@ static int vfs2_shm_map(sqlite3_file *file,
 	return rv;
 }
 
+static __attribute__((noinline)) int busy(void)
+{
+	return SQLITE_BUSY;
+}
+
 static int vfs2_shm_lock(sqlite3_file *file, int ofst, int n, int flags)
 {
-	struct vfs2_file *xfile = (struct vfs2_file *)file;
+	struct file *xfile = (struct file *)file;
+	struct entry *e = xfile->entry;
 
 	assert(file != NULL);
 	assert(ofst >= 0);
```

Comment:
lines 918-919 and 921-922 check the same conditions.

Reply:
Good point, fixed